src/NuGet.Clients/VsConsole/PowerShellHost/PowerShellHost.cs

```diff
@@ -372,95 +375,174 @@ private void UpdateWorkingDirectory()
-                            sortedPackages.AddRange(installedPackages);
-                            packagesFolderPackages.UnionWith(installedPackages);
+                            // Index packages.config references by target framework since this affects dependencies
+                            NuGetFramework targetFramework;
+                            if (!project.TryGetMetadata(NuGetProjectMetadataKeys.TargetFramework, out targetFramework))
+                            {
+                                targetFramework = NuGetFramework.AnyFramework;
+                            }
+
+                            HashSet<PackageIdentity> fwPackages;
+                            if (!packagesConfigInstalled.TryGetValue(targetFramework, out fwPackages))
+                            {
+                                fwPackages = new HashSet<PackageIdentity>();
+                                packagesConfigInstalled.Add(targetFramework, fwPackages);
+                            }
+
+                            fwPackages.UnionWith(installedRefs.Select(reference => reference.PackageIdentity));
                         }
                     }
                 }
 
-                // Get the path to the Packages folder.
-                var packagesFolderPath = packageManager.PackagesFolderSourceRepository.PackageSource.Source;
-                var packagePathResolver = new PackagePathResolver(packagesFolderPath);
+                // Each id/version should only be executed once
+                var finishedPackages = new HashSet<PackageIdentity>();
+
+                // Packages.config projects
+                if (packagesConfigInstalled.Any())
+                {
+                    await ExecuteInitPs1ForPackagesConfig(
+                        packageManager,
+                        packagesConfigInstalled,
+                        finishedPackages);
+                }
+
+                // build integrated projects
+                if (sortedGlobalPackages.Any())
+                {
+                    ExecuteInitPs1ForBuildIntegrated(
+                        sortedGlobalPackages,
+                        finishedPackages);
+                }
+            }
+        }
+
+        private async Task ExecuteInitPs1ForPackagesConfig(
+            NuGetPackageManager packageManager,
+            Dictionary<NuGetFramework, HashSet<PackageIdentity>> packagesConfigInstalled,
+            HashSet<PackageIdentity> finishedPackages)
+        {
+            // Get the path to the Packages folder.
+            var packagesFolderPath = packageManager.PackagesFolderSourceRepository.PackageSource.Source;
+            var packagePathResolver = new PackagePathResolver(packagesFolderPath);
 
-                var nugetPaths = NuGetPathContext.Create(_settings);
-                var fallbackResolver = new FallbackPackagePathResolver(nugetPaths);
+            var packagesToOrder = new HashSet<ResolverPackage>();
```

Comment:
packagesDependencies is pretty vague. The flow here is packagesToOrder and then it is ordered into sortedPackages. It could be called packagesToSort I guess to more closely match the result.